### ISSUE TITLE
Bug: index.js breaks all plugins

### ISSUE THREAD

**Issue by yandodov** [getkirby/editor#60]
Something very weird. I cloned the editor and put it in a project. When I opened the panel, all custom fields were broken and there was an error in the console:

```
index.js?1565433242:37 Uncaught Error: Cannot find module 'function () {function ld(t){return uk(t)||vk(t)||tg()}function tg(){throw new TypeError("Invalid attempt to spread non-iterable instance")}function vk(t){if(Symbol.iterator in Object(t)||"[object Arguments]"===Object.prototype.toString.call(t))return Array.from(t)}function uk(t){if(Array.isArray(t)){for(var e=0,n=new Array(t.length);e<t.length;e++)n[e]=t[e];return n}}function mf(t){for(var e=1;e<arguments.length;e++){var n=null!=arguments[e]?arguments[e]:{},o=Object.keys(n);"function"==typeof Object.getOwnPropertySymbols&&(o=o.concat(Object.getOwnPropertySymbols(n).filter(function(t){return Object.getOwnPropertyDescriptor(n,t).enumerable}))),o.forEach(function(e){rk(t,e,n[e])})}return t}function rk(t,e,n){return e in t?Object.defineProperty(t,e,{value:n,enumerable:!0,configurable:!0,writable:!0}):t[e]=n,t}var Ze={h1:function(t){return{type:"h1",content:t.innerHTML}},h2:function(t){return{type:"h2",content:t.innerHTML}},h3:function(t){return{type:"h3",content:t.innerHTML}},blockquote:function(t){return{type:"blockquote",content:t.innerHTML}},hr:function(){return{type:"hr"}},p:function(t){return 0!==t.innerText.length&&{type:"paragraph",content:t.innerHTML}},br:function(){return!1},img:function(t){return{type:"image",attrs:{alt:t.getAttribute("alt"),src:t.getAttribute("src")}}},pre:function(t){var e=t.querySelector("code");if(!e)return{type:"paragraph",content:t.innerHTML};var n=e.getAttribute("class"),r=null;return n.match(/^language-([a-z_-]+)$/)&&(r=n.replace("language-","")),{type:"code",content:e.innerHTML,attrs:{language:r}}},li:function(t){return{type:t.parentElement&&"OL"===t.parentElement.tagName?"ol":"ul",content:t.innerHTML}},tr:function(t){return{type:"paragraph",content:t.innerHTML}},figure:function(t){var e=t.querySelector("img"),n=t.querySelector("iframe"),r=t.querySelector("figcaption");return e?{type:"image",attrs:{alt:e.getAttribute("alt"),src:e.getAttribute("src"),caption:r?r.innerText:null}}:{type:"video",attrs:{src:n.getAttribute("src"),caption:r?r.innerText:null}}}};var db={props:{blocks:Object,block:Object,menu:{type:Function,default:function(){return function(){return[]}}}},data:function(){return{mode:null,menuItems:[]}},methods:{open:function(){this.menuItems=this.menu(),this.$refs.blockOptions.toggle()},close:function(){this.menuItems=[],this.$refs.blockOptions.close()},go:function(t){this.mode=t,this.$refs.blockOptions.open()},onClose:function(){this.mode=null}}};if(typeof db==="function"){db=db.options}Object.assign(db,function(){var render=function(){var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c("nav",{staticClass:"k-editor-block-options"},[_c("button",{staticClass:"k-editor-block-option k-sort-handle",attrs:{"type":"button"}},[_c("k-icon",{attrs:{"type":"sort"}})],1),_vm._v(" "),_c("k-dropdown",[_c("button",{staticClass:"k-editor-block-option k-editor-block-options-sort",attrs:{"type":"button"},on:{"mousedown":function($event){$event.stopPropagation();return _vm.open($event)}}},[_c("k-icon",{attrs:{"type":"angle-down"}})],1),_vm._v(" "),_c("k-dropdown-content",{ref:"blockOptions",staticClass:"k-editor-block-option-dropdown",on:{"close":_vm.onClose},nativeOn:{"mousedown":function($event){$event.stopPropagation()}}},[_vm.mode==="insert"?[_c("k-dropdown-item",{staticClass:"k-editor-block-option-heading",attrs:{"icon":"angle-left"},on:{"click":function($event){return _vm.go()}}},[_vm._v(" "+_vm._s(_vm.$t("editor.options.insert.below"))+" \u2026 ")]),_vm._v(" "),_c("hr"),_vm._v(" "),_vm._l(_vm.blocks,function(definition){return _c("k-dropdown-item",{key:definition.type,attrs:{"icon":definition.icon},on:{"click":function($event){return _vm.$emit("add",definition.type)}}},[_vm._v(" "+_vm._s(definition.label)+" ")])})]:_vm.mode==="convert"?[_c("k-dropdown-item",{staticClass:"k-editor-block-option-heading",attrs:{"icon":"angle-left"},on:{"click":function($event){return _vm.go()}}},[_vm._v(" "+_vm._s(_vm.$t("editor.options.convert"))+" \u2026 ")]),_vm._v(" "),_c("hr"),_vm._v(" "),_vm._l(_vm.blocks,function(definition){return _vm.block.type!==definition.type?_c("k-dropdown-item",{key:definition.type,attrs:{"icon":definition.icon},on:{"click":function($event){return _vm.$emit("convert",definition.type)}}},[_vm._v(" "+_vm._s(definition.label)+" ")]):_vm._e()})]:[_c("k-dropdown-item",{attrs:{"icon":"add"},on:{"click":function($event){return _vm.go("insert")}}},[_vm._v(_vm._s(_vm.$t("editor.options.insert.below"))+" \u2026")]),_vm._v(" "),_c("hr"),_vm._v(" "),_c("k-dropdown-item",{attrs:{"icon":"refresh"},on:{"click":function($event){return _vm.go("convert")}}},[_vm._v(_vm._s(_vm.$t("editor.options.convert"))+" \u2026")]),_vm._v(" "),_c("k-dropdown-item",{attrs:{"icon":"copy"},on:{"click":function($event){return _vm.$emit("duplicate")}}},[_vm._v(_vm._s(_vm.$t("editor.options.duplicate")))]),_vm._v(" "),_c("hr"),_vm._v(" "),_vm.menuItems.length?[_vm._l(_vm.menuItems,function(menuItem){return _c("k-dropdown-item",{attrs:{"icon":menuItem.icon,"index.js?1565433242:37)
    at newRequire (index.js?1565433242:21)
    at index.js?1565433242:15368
```

I saw that one of my compiled plugin scripts began with `!` and I prepended it in the Editor _index.js_ as well (it wasn't there). After that, everything started working again.

This is obviously not a bug with the Editor, but with Parcel, I guess. Although, I've had the same issue before while I was developing a plugin and its JavaScript wasn't compiled (dev mode). I was using Webpack for it, which means it might not be Parcel then? Kirby joins all plugin scripts in a single file. Perhaps that's where things break?

**Edit:** @bastianallgeier I think this might be a Kirby issue. Perhaps it should wrap plugin scripts in an IIFE to avoid such problems. By the way, is there a reason not to have a separate `<script>` tag for each plugin script? Why concatenate them in one large file?

**Comment by mauricerenck** [comment]
Can confirm that.
Had the same issue with one of my own plugins. Using editor causes that all the plugins sections aren't working anymore.

**Comment by bastianallgeier** [comment]
This is actually an issue in Kirby, which will be fixed in 3.2.4
